module2/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,59 @@
+#include "compression.hpp"
+
+#include <algorithm>
+#include <cctype>
+#include <iostream>
+
+SqueezedMap compressGrayscale(DiffusedMap& getGray)
+{
+    SqueezedMap squeezedMap;
+    squeezedMap.reserve(width * height);
+
+    std::transform(getGray.begin(), getGray.end(), getGray.begin(),
+
+                   [&squeezedMap](std::array<uint8_t, width> everyRow) {
+                       auto firstIt = everyRow.begin();
+                       while (firstIt != everyRow.end()) {
+                           auto secondIt = std::find_if_not(firstIt, everyRow.end(), [firstIt](int x) { return x == *firstIt; });
+                           squeezedMap.emplace_back(*firstIt, std::distance(firstIt, secondIt));
+                           firstIt = secondIt;
+                       }
+
+                       return everyRow;
+                   });
```

Comment:
lloking for iterators in std:;tranform I assume you transform `getGray` vector. But inside funtion you modify a squeezedMap.  This is not good practise. Please us std::for_each() or range loop.